packages/edit-site/src/components/global-styles/typography-panel.js

```diff
@@ -1,419 +1,43 @@
-					isShownByDefault
-				>
-					<FontSizePicker
-						value={ fontSize }
-						onChange={ setFontSize }
-						fontSizes={ fontSizes }
-						disableCustomFontSizes={ disableCustomFontSizes }
-						withReset={ false }
-						withSlider
-						size="__unstable-large"
-						__nextHasNoMarginBottom
-					/>
-				</ToolsPanelItem>
-			) }
-			{ hasAppearanceControl && (
-				<ToolsPanelItem
-					className="single-column"
-					label={ appearanceControlLabel }
-					hasValue={ hasFontAppearance }
-					onDeselect={ resetFontAppearance }
-					isShownByDefault
-				>
-					<FontAppearanceControl
-						value={ {
-							fontStyle,
-							fontWeight,
-						} }
-						onChange={ ( {
-							fontStyle: newFontStyle,
-							fontWeight: newFontWeight,
-						} ) => {
-							setFontStyle( newFontStyle );
-							setFontWeight( newFontWeight );
-						} }
-						hasFontStyles={ hasFontStyles }
-						hasFontWeights={ hasFontWeights }
-						size="__unstable-large"
-						__nextHasNoMarginBottom
-					/>
-				</ToolsPanelItem>
-			) }
-			{ hasLineHeightEnabled && (
-				<ToolsPanelItem
-					className="single-column"
-					label={ __( 'Line height' ) }
-					hasValue={ hasLineHeight }
-					onDeselect={ resetLineHeight }
-					isShownByDefault
-				>
-					<LineHeightControl
-						__nextHasNoMarginBottom
-						__unstableInputWidth="auto"
-						value={ lineHeight }
-						onChange={ setLineHeight }
-						size="__unstable-large"
-					/>
-				</ToolsPanelItem>
-			) }
-			{ hasLetterSpacingControl && (
-				<ToolsPanelItem
-					className="single-column"
-					label={ __( 'Letter spacing' ) }
-					hasValue={ hasLetterSpacing }
-					onDeselect={ resetLetterSpacing }
-					isShownByDefault
-				>
-					<LetterSpacingControl
-						value={ letterSpacing }
-						onChange={ setLetterSpacing }
-						size="__unstable-large"
-						__unstableInputWidth="auto"
-					/>
-				</ToolsPanelItem>
-			) }
-			{ hasTextTransformControl && (
-				<ToolsPanelItem
-					label={ __( 'Letter case' ) }
-					hasValue={ hasTextTransform }
-					onDeselect={ resetTextTransform }
-					isShownByDefault
-				>
-					<TextTransformControl
-						value={ textTransform }
-						onChange={ setTextTransform }
-						showNone
-						isBlock
-						size="__unstable-large"
-						__nextHasNoMarginBottom
-					/>
-				</ToolsPanelItem>
-			) }
-			{ hasTextDecorationControl && (
-				<ToolsPanelItem
-					className="single-column"
-					label={ __( 'Text decoration' ) }
-					hasValue={ hasTextDecoration }
-					onDeselect={ resetTextDecoration }
-					isShownByDefault
-				>
-					<TextDecorationControl
-						value={ textDecoration }
-						onChange={ setTextDecoration }
-						size="__unstable-large"
-						__unstableInputWidth="auto"
-					/>
-				</ToolsPanelItem>
-			) }
-		</ToolsPanel>
+		<StylesTypographyPanel
+			inherit={ inheritedStyle }
+			value={ style }
+			onChange={ setStyle }
+			name={ name }
+			element={ element === 'heading' ? headingLevel : element }
```

Comment:
Not sure that the `wp.blockEditor.TypographyPanel` should care about `name` and `element`? Ideally it should take a style and settings object and not mind where those objects came from.

Moreover, does selecting a heading level make sense in the block inspector? I imagine that the heading level selector needs to be moved from `wp.blockEditor.TypographyPanel` to `wp.editSite.TypographyPanel`.

Reply:
> Not sure that the wp.blockEditor.TypographyPanel should care about name and element? Ideally it should take a style and settings object and not mind where those objects came from.

Yeah, this is one of the remaining points above, either we add a "settings" prop which would allow us to know what features are enabled/disabled or we have a dependency to GlobalStylesContext.

I do like the settings prop better but that forces us to define what that object look like and how to "build" it based on theme.json (for both edit-site and block-editor)

Reply:
I've added the "settings" prop to avoid the "GlobalStylesContext" dependency but I've kept name, element for now that are used to check "block supports". In theory we could merge the global styles settings with the block supports and check only one place in the component itself, but for now I'm leaving this for a later stage. The settings prop as it is now should be enough to allow the component to be used in the inspector controls as well.

Reply:
A `GlobalStylesContext` dependency would be more bearable imo if we combine `GlobalStylesContext` and `BlockEditorProvider` which I think makes sense anyway.

A `settings` prop is nice though for cases where the enabled features depend on the selected block's attributes. 

Could you spin up a new PR stacked on top of this one that uses `TypographyPanel` in the block inspector? I think it will inform a lot of these decisions 😀

Reply:
> Could you spin up a new PR stacked on top of this one that uses TypographyPanel in the block inspector? I think it will inform a lot of these decisions 😀

I've went ahead and actually merged that into the current PR. I don't see the value of the current PR without the component used in both places. So here's the commit for that part [19e0eee](https://github.com/WordPress/gutenberg/pull/47356/commits/19e0eee9c31550dc4bfdb0b21ef87c9ba454cd44) including some small fixes for the TypographyPanel component.

To be honest, I think that commit speaks to itself and that now I'm totally convinced that this is the way to go for all our "styles" panels.